app/Project.php

```diff
@@ -145,4 +145,84 @@ public function user()
     {
         return $this->belongsTo(User::class);
     }
+
+    /**
+     *
+     *
+     * @return 
+     */
+    public function followers()
+    {
+        return $this->hasMany(ProjectFollower::class);
+    }
+
+    /**
+     * Get the count of followers for the project
+     *
+     * @return
+     */
+    public function getFollowerCount()
+    {
+        return count(ProjectFollower::where('project_id', '=', $this->id)->get());
+    }
+
+    /**
+     * Check if user is a follower of this project
+     *
+     * @return
+     */
+    public function isUserFollower()
```

Comment:
This function signature is weird to me. I keep wanting to see a user passed as a parameter to check. If you did make it accept a user, you'd get the benefit of being able to check arbitrary users for free, which may be nice in the future. But idk how you'd handle auth in that case.